packages/proto-build/src/transformers.rs

```diff
@@ -436,92 +436,67 @@ fn append_type_path(path: &str, name: &str) -> String {
 
 fn has_repr_attribute(attrs: &[Attribute]) -> bool {
     attrs.iter().any(|attr| {
-        // Parse the attribute's meta information.
-        let meta = match attr.parse_meta() {
-            Ok(meta) => meta,
-            Err(_) => return false,
-        };
-
-        // Check if the attribute name is "prost".
-        match meta {
-            syn::Meta::List(list) => list
-                .path
-                .get_ident()
-                .filter(|ident| ident.to_string() == "repr")
-                .is_some(),
-
-            _ => return false,
+        // Check if the attribute name is "repr".
+        if attr.path().is_ident("repr") {
+            return true;
         }
+
+        false
     })
 }
 
 /// Searches for the "enumeration" attribute in an attribute list of the form
 /// #[prost(enumeration = "target" )] and returns the target value if it does exist.
 fn find_prost_enumeration_value(attrs: &[Attribute]) -> Option<String> {
     attrs.iter().find_map(|attr| {
-        // Parse the attribute's meta information.
-        let meta = match attr.parse_meta() {
-            Ok(meta) => meta,
-            Err(_) => return None,
-        };
-
         // Check if the attribute name is "prost".
-        let list = match meta {
-            syn::Meta::List(list)
-                if list
-                    .path
-                    .get_ident()
-                    .filter(|ident| ident.to_string() == "prost")
-                    .is_some() =>
-            {
-                list
+        if !attr.path().is_ident("prost") {
+            return None;
+        }
+
+        // Search all nested attributes and look for the "enumeration" attribute, then getting its value.
+        let mut enumeration_value = None::<String>;
+        attr.parse_nested_meta(|meta|{
+            if !meta.path.is_ident("enumeration") {
+                return Ok(());
             }
-            _ => return None,
-        };
-
-        // Search all nested attributes and look for the "enumeration" attribute.
-        list.nested.iter().find_map(|nested| {
-            if let syn::NestedMeta::Meta(syn::Meta::NameValue(nv)) = nested {
-                if nv.path.is_ident("enumeration") {
-                    if let syn::Lit::Str(s) = &nv.lit {
-                        return Some(s.value());
-                    }
-                }
+
+            let value = meta.value()?;
+            let s : syn::LitStr = value.parse()?;
+            enumeration_value = Some(s.value());
+            return Ok(());
+        }).unwrap_or_else(|e|{
+            // Do nothing if the error is "expected `,`"
+            if e.to_string() != "expected `,`".to_string() {
+                panic!("{}", e)
             }
```

Comment:
The error with message "expected `,` happens, but it works properly to transform to generated rust into cosmwasm version. So it can be safely filtered.

Reply:
We might remove the specific filter after it is solved:
https://github.com/dtolnay/syn/issues/1448